File: leetcode/h_index.cpp
using namespace std;
#include <iostream>
#include <vector>
#include <algorithm>


class Solution {
public: 
    int hIndex(vector<int>& citations) {
        sort(citations.begin(), citations.end());
        int n = citations.size();
        int hIndex = 0;
        for (int i = 0; i < n; i++) {
            int h = min(citations[i], n - i);
            hIndex = max(hIndex, h);
        }
        return hIndex;
    }
};

int main() {
    Solution s;
    vector<int> citations = {3, 0, 6, 1, 5};
    cout << s.hIndex(citations) << endl;
}

File: leetcode/HIndex.java
import java.util.Arrays;


public class HIndex {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        int n = citations.length;
        int hIndex = 0;
        for (int i = 0; i < n; i++) {
            int h = Math.min(citations[i], n - i);
            hIndex = Math.max(hIndex, h);
        }
        return hIndex;
    }

    public static void main(String[] args) {
        HIndex hIndex = new HIndex();
        int[] citations = {3, 0, 6, 1, 5};
        System.out.println(hIndex.hIndex(citations));
    }
}